### PR TITLE
MULE-13495 Foreach gets stuck with empty Paging Provider

### PR DESCRIPTION
**Pull request by dfeist**

### CODE REVIEW

File: src/test/java/org/mule/extension/db/integration/select/SelectTestCase.java [hunk 1]
```diff
@@ -83,6 +83,11 @@ public void maxRows() throws Exception {
   }
 
   @Test
+  public void emptyResult() throws Exception {
+    flowRunner("emptyResult").run().getMessage();
+  }
```
Comment (dfeist) [inline src/test/java/org/mule/extension/db/integration/select/SelectTestCase.java]: @pablomusumeci Can we add any other assertions here?

File: src/test/resources/integration/select/select-config.xml [hunk 2]
```diff
@@ -46,6 +46,15 @@
         </db:select>
     </flow>
 
+    <flow name="emptyResult">
+        <db:select config-ref="dbConfig">
```
Comment (marianogonzalez) [inline src/test/resources/integration/select/select-config.xml]: This test is highly dependant on nobody else adding records into SPACESHIP table for some other purpose and inadvertently making this test useless.

This should start by truncating the table.

File: src/test/java/org/mule/extension/db/integration/select/SelectTestCase.java (hunk 1 above)
Comment (marianogonzalez) [reply src/test/java/org/mule/extension/db/integration/select/SelectTestCase.java]: you can take the returned iterator and verify that hasNext() is false

File: src/test/java/org/mule/extension/db/integration/select/SelectTestCase.java [hunk 3]
```diff
@@ -87,6 +87,7 @@ public void maxRows() throws Exception {
 
   @Test
   public void emptyResult() throws Exception {
+    executeWithDataSources(dataSource -> testDatabase.truncateSpaceshipTable(dataSource.getConnection()));
```
Comment (dfeist) [inline src/test/java/org/mule/extension/db/integration/select/SelectTestCase.java]: I deleted it in XML, so maybe that can be removed, if you do it here?

Comment (pablomusumeci) [reply src/test/java/org/mule/extension/db/integration/select/SelectTestCase.java]: Yes, I told Mariano that I hadn't seen the deletion in the flow. I'm removing that from the xml right now

**Pull request merged**